pkg/util/pravegacluster.go

```diff
@@ -64,17 +64,25 @@ func HealthcheckCommand(port int32) []string {
 }
 
-func ControllerReadinessCheck(port int32, authflag bool, tlsflag bool) []string {
-	if authflag == true && tlsflag == true {
-		return []string{"/bin/sh", "-c", fmt.Sprintf("curl -v -k -u testtls:testtls -s -X GET 'https://localhost:%d/v1/scopes/' 2>&1 -H 'accept: application/json' | grep 401", port)}
-	}
+//This function check for the readiness of the controller in the following cases
+//1) Auth and TLS Enabled- in this case, we check if the controller is properly enabled with authentication or not and we do a get on controller and with dummy credentials(testtls:testtls) and the controller returns 401 error in this case if it's correctly configured
+//2) Auth Enabled and TLS Disabled- in this case, we check if the controller is properly enabled with authentication or not and we do a get on controller and with dummy credentials(testtls:testtls) and the controller returns 401 error in this case if it's correctly configured
+//3) Auth Disabled and TLS Enabled- in this case, we check if the controller can create scopes or not by checking if _system scope is present or not
+//4) Auth and TLS Disabled- in this case, we check if the controller can create scopes or not by checking if _system scope is present or not
+func ControllerReadinessCheck(port int32, authflag bool) []string {
+	// This is to check the readiness of controller in case auth is Enabled
+	// here we are using login credential as testtls:testtls which should
+	// not be used as auth credential and we depend on controller giving us
+	// 401 error which means controller is properly configured with auth
+	// it checks both cases when tls is enabled as well as tls disabled
+	// with auth enabled
 	if authflag == true {
-		return []string{"/bin/sh", "-c", fmt.Sprintf("curl -v -k -u testauth:testauth -s -X GET 'http://localhost:%d/v1/scopes/' 2>&1 -H 'accept: application/json' | grep 401", port)}
-	}
-	if tlsflag == true {
-		return []string{"/bin/sh", "-c", fmt.Sprintf("curl -v -k -s -X GET 'https://localhost:%d/v1/scopes/' 2>&1 -H 'accept: application/json' | grep '_system'", port)}
+		return []string{"/bin/sh", "-c", fmt.Sprintf("echo $JAVA_OPTS | grep 'controller.auth.tlsEnabled=true' &&  curl -v -k -u testtls:testtls -s -X GET 'https://localhost:%d/v1/scopes/' 2>&1 -H 'accept: application/json' | grep 401 || (echo $JAVA_OPTS | grep 'controller.auth.tlsEnabled=false' && curl -v -k -u testtls:testtls -s -X GET 'http://localhost:%d/v1/scopes/' 2>&1 -H 'accept: application/json' | grep 401 ) ||  (echo $JAVA_OPTS | grep 'controller.security.tls.enable=true' && echo $JAVA_OPTS | grep -v 'controller.auth.tlsEnabled' && curl -v -k -u testtls:testtls -s -X GET 'https://localhost:%d/v1/scopes/' 2>&1 -H 'accept: application/json' | grep 401 ) || (curl -v -k -u testtls:testtls -s -X GET 'http://localhost:%d/v1/scopes/' 2>&1 -H 'accept: application/json' | grep 401 )", port, port, port, port)}
```

Comment:
I understand the intent about inducing 401 errors to validate that the rest api is responsive and secured but...  the big downside is that it is flooding the controller logs with authentication errors that are very very confusing and making it appear like there are some authentication issues in the system.

Wondering if there's a better way to achieve this....

Reply:
Currently, this is the best we could do from the operator side and this is a temporary fix which we will change once we have api's that we can call for doing health check at the pravega side link to the pravega issue is https://github.com/pravega/pravega-operator/issues/357

Reply:
Understood thank you.